dali/test/python/test_operator_coco_reader.py

```diff
@@ -25,44 +25,60 @@
 
 test_data = {
-    'car-race-438467_1280.jpg' : 0,
-    'clock-1274699_1280.jpg' : 5,
-    'kite-1159538_1280.jpg' : 6,
-    'cow-234835_1280.jpg' : 17,
-    'home-office-336378_1280.jpg' : 21,
-    'suit-2619784_1280.jpg' : 39,
-    'business-suit-690048_1280.jpg' : 41,
-    'car-604019_1280.jpg' : 59
+    'car-race-438467_1280.jpg' : 17,
+    'clock-1274699_1280.jpg' : 6,
+    'kite-1159538_1280.jpg' : 21,
+    'cow-234835_1280.jpg' : 59,
+    'home-office-336378_1280.jpg' : 39,
+    'suit-2619784_1280.jpg' : 0,
+    'business-suit-690048_1280.jpg' : 5,
+    'car-604019_1280.jpg' : 41
 }
 
 images = list(test_data.keys())
 expected_ids = list(test_data.values())
 
-def test_operator_coco_reader():
-    with tempfile.TemporaryDirectory() as annotations_dir:
-        pipeline = Pipeline(batch_size=2, num_threads=4, device_id=0)
-        with pipeline:
-            inputs, _, _, ids = fn.coco_reader(
-                file_root=file_root,
-                annotations_file=train_annotations,
-                image_ids=True,
-                images=images,
-                save_preprocessed_annotations=True,
-                save_preprocessed_annotations_dir=annotations_dir)
-            pipeline.set_outputs(ids)
-        pipeline.build()
-
-        i = 0
-        while i < len(images):
-            out = pipeline.run()
-            assert out[0].at(0) == expected_ids[i]
-            assert out[0].at(1) == expected_ids[i + 1]
-            i = i + 2
-
-        filenames_file = os.path.join(annotations_dir, 'filenames.dat')
-        with open(filenames_file) as f:
-            lines = f.read().splitlines()
-        assert lines.sort() == images.sort()
+def test_operator_coco_reader_custom_order():
+    custom_orders = [
+        None,  # natural order
+        [0, 2, 4, 6, 1, 3, 5, 7],  # altered order
+        [0, 1, 2, 3, 2, 1, 4, 1, 5, 2, 6, 7],  # with repetitions
+        ]
+
+    def check_operator_coco_reader_custom_order(order=None):
+        if not order:
+            order = range(len(test_data))
+        keys = list(test_data.keys())
+        values = list(test_data.values())
+        images = [keys[i] for i in order]
+        expected_ids = [values[i] for i in order]
+        with tempfile.TemporaryDirectory() as annotations_dir:
+            pipeline = Pipeline(batch_size=2, num_threads=4, device_id=0)
+            with pipeline:
+                inputs, _, _, ids = fn.coco_reader(
+                    file_root=file_root,
+                    annotations_file=train_annotations,
+                    image_ids=True,
+                    images=images,
+                    save_preprocessed_annotations=True,
+                    save_preprocessed_annotations_dir=annotations_dir)
+                pipeline.set_outputs(ids)
+            pipeline.build()
+
+            i = 0
+            while i < len(images):
+                out = pipeline.run()
+                assert out[0].at(0) == expected_ids[i]
```

Comment:
Maybe just put here another loop that iterates batch_size times instead of hardcoding the repetition two times?